app/api/v1/endpoints/strategies.py

```
@@ -43,6 +45,39 @@
+        pipe.zcard(rate_key)
+        pipe.expire(rate_key, window)
+        results = await pipe.execute()
+
+        current_count = results[2]
+
+        if current_count > limit:
+            raise HTTPException(
+                status_code=429,
+                detail=f"Rate limit exceeded. Max {limit} requests per {window} seconds.",
+                headers={"Retry-After": str(window)}
+            )
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)